backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/monitoring/aspect/MonitorAspect.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.servlet.HandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+@Aspect
+@Component
+public class MonitorAspect
+{
+	private final MicrometerPerformanceMonitoringService service;
+	private static final String PERF_MON_SYSCONFIG_NAME = "de.metas.monitoring.annotation.enable";
+	private static final boolean SYS_CONFIG_DEFAULT_VALUE = false;
+	private final ISysConfigBL SYS_CONFIG_BL = Services.get(ISysConfigBL.class);
+
+
+	public MonitorAspect(@NonNull final MicrometerPerformanceMonitoringService service)
+	{
+		this.service = service;
+	}
+
+	@Around("execution(* *(..)) && @annotation(de.metas.monitoring.annotation.Monitor)")
+	public Object monitorMethod(ProceedingJoinPoint pjp) throws Throwable
+	{
+		final boolean perfMonIsActive = SYS_CONFIG_BL.getBooleanValue(PERF_MON_SYSCONFIG_NAME, SYS_CONFIG_DEFAULT_VALUE);
+		if(!perfMonIsActive)
+		{
+			return pjp.proceed();
+		}
+
+		final Callable<?> callable = wrapAsCallable(pjp);
+		final PerformanceMonitoringService.Metadata metadata;
+
+		final Method method = ((MethodSignature) pjp.getSignature()).getMethod();
+		final Monitor monitorAnnotation = method.getAnnotation(Monitor.class);
+
+		if(monitorAnnotation.type() == PerformanceMonitoringService.Type.REST_CONTROLLER_WITH_WINDOW_ID)
+		{
+			metadata = PerformanceMonitoringService.Metadata.builder()
+					.className(pjp.getTarget().getClass().getSimpleName())
+					.type(monitorAnnotation.type())
+					.functionName(method.getName())
+					.windowNameAndId(getWindowNameAndId())
+					.build();
+		}
+		else
+		{
+			metadata = PerformanceMonitoringService.Metadata.builder()
+					.className(pjp.getTarget().getClass().getSimpleName())
+					.type(monitorAnnotation.type())
+					.functionName(method.getName())
+					.build();
+		}
+
+		return service.monitor(callable, metadata);
+
+	}
+
+	private Callable<Object> wrapAsCallable(final ProceedingJoinPoint pjp)
+	{
+		final Callable<Object> callable = new Callable<Object>()
+		{
+
+			@Override
+			public Object call() throws Exception
+			{
+				try
+				{
+					return pjp.proceed();
+				}
+				catch (final Exception e)
+				{
+					throw e;
+				}
+				catch (final Throwable t)
+				{
+					throw AdempiereException.wrapIfNeeded(t);
+				}
+			}
+
+			@Override
+			public String toString()
+			{
+				return pjp.getTarget().toString();
+			}
+		};
+
+		return callable;
+	}
+
+	private String getWindowNameAndId(){
+		final HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+		final Map pathVariables = (Map) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+		final String windowId = (String)pathVariables.get("windowId");
+		final IADWindowDAO iadWindowDAO = Services.get(IADWindowDAO.class);
```

Comment:
if u cannot extract this service as a field, pls add a comment here shortly explaining why